modules/enterprise-management/src/test/java/com/ibm/cloud/platform_services/enterprise_management/v1/EnterpriseManagementIT.java

```diff
@@ -1,433 +1,563 @@
-        JSONObject obj = new JSONObject(response);
-        accountId = obj.getString("id");
-    }
-
-    @Test
-    public void test21_GetAccountByAccountIdLastOperation() throws Throwable {
-        accountManagementService.getAccountByAccountIdLastOperation(accountId, token, config.get("AM_HOST"));
-    }
-
-    @Test
-    public void test22_GetAccountActivationToken() throws Throwable {
-        URL url = null;
-        try {
-            url = new URL(config.get("DB_URL"));
-        } catch (MalformedURLException e) {
-            fail("The url is not well formed: " + url);
+        ListEnterprisesResponse listEnterprisesResponseResult = response.getResult();
+        assertNotNull(listEnterprisesResponseResult);
+
+        nextDocId = getNextDocId(listEnterprisesResponseResult.getNextUrl());
+        enterpriseList.addAll(listEnterprisesResponseResult.getResources());
+
+        log(String.format("Received a total of %d enterprises.%n", enterpriseList.size()));
+      } while (nextDocId != null);
+    } catch (ServiceResponseException e) {
+      fail(String.format("Service returned status code %d: %s%nError details: %s", e.getStatusCode(), e.getMessage(),
+          e.getDebuggingInfo()));
+    }
+  }
+
+  @Test
+  public void testGetEnterprise() throws Exception {
+    try {
+      GetEnterpriseOptions getEnterpriseOptions = new GetEnterpriseOptions.Builder()
+          .enterpriseId(enterpriseId)
+          .build();
+
+      // Invoke operation
+      Response<Enterprise> response = service
+          .getEnterprise(getEnterpriseOptions)
+          .execute();
+      // Validate response
+      assertNotNull(response);
+      assertEquals(response.getStatusCode(), 200);
+
+      Enterprise enterpriseResult = response.getResult();
+
+      assertNotNull(enterpriseResult);
+    } catch (ServiceResponseException e) {
+      fail(String.format("Service returned status code %d: %s%nError details: %s", e.getStatusCode(), e.getMessage(),
+          e.getDebuggingInfo()));
+    }
+  }
+
+  @Test
+  public void testUpdateEnterprise() throws Exception {
+    try {
+      UpdateEnterpriseOptions updateEnterpriseOptions = new UpdateEnterpriseOptions.Builder()
+          .enterpriseId(enterpriseId)
+          .name(updatedExampleEnterpriseName)
+          .primaryContactIamId(accountIamId)
+          .build();
+
+      // Invoke operation
+      Response<Void> response = service
+          .updateEnterprise(updateEnterpriseOptions)
+          .execute();
+      // Validate response
+      assertNotNull(response);
+      assertEquals(response.getStatusCode(), 204);
+    } catch (ServiceResponseException e) {
+      fail(String.format("Service returned status code %d: %s%nError details: %s", e.getStatusCode(), e.getMessage(),
+          e.getDebuggingInfo()));
+    }
+  }
+
+  @AfterClass
+  public void tearDown() {
+    // Add any clean up logic here
+    System.out.println("Clean up complete.");
+  }
+
+  private String getNextDocId(String s) {
```

Comment:
Instead of this function, please use the java core's UrlHelper.getQueryParam() function.